### PR TITLE
fix:Error in throwing null parameter in php8.1.

### PR DESCRIPTION
**Pull request by StarsRivers**

### CODE REVIEW

**Comment by jaideraf** [comment]
Fixes https://github.com/SemanticMediaWiki/SemanticMediaWiki/issues/5521
Fixes https://github.com/SemanticMediaWiki/SemanticMediaWiki/issues/5514

File: src/Utils/TemplateEngine.php [hunk 1]
```diff
@@ -116,9 +116,13 @@ public function compile( $target, array $args = [] ) {
 
 		$complied = $this->container[$target];
 
-		foreach ( $args as $key => $value ) {
-			$complied = str_replace( [ '{{' . $key . '}}', '{{#' . $key . '}}' ], $value, $complied );
-		}
+		
+		foreach ($args as $key => $value) {
```
Comment (Nikerabbit) [inline src/Utils/TemplateEngine.php:120]: Can you keep the existing code style and use tabs for indendation?

Comment (JeroenDeDauw) [reply src/Utils/TemplateEngine.php:120]: Marked as unresolved since it's not resolved

Comment (JeroenDeDauw) [reply src/Utils/TemplateEngine.php:119]: Not resolved. 

Note the spaces

```php
foreach ( $args as $key => $value ) {
```

File: src/Utils/TemplateEngine.php [hunk 2]
```diff
@@ -116,9 +116,9 @@ public function compile( $target, array $args = [] ) {
 
 		$complied = $this->container[$target];
 		
-		foreach ($args as $key => $value) {
-			if ($value !== null) {
-				$complied = str_replace(['{{' . $key . '}}', '{{#' . $key . '}}'], $value, $complied);
+		foreach ( $args as $key => $value) {
+			if ( $value !== null) {
+				$complied = str_replace( ['{{' . $key . '}}', '{{#' . $key . '}}'], $value, $complied);
```
Comment (jaideraf) [inline src/Utils/TemplateEngine.php:121]: @StarsRivers the spaces must be present in both sides of the code, no only in the left side

**Pull request merged**